src/common/feedGenerator.ts

```diff
@@ -670,14 +700,44 @@ export const bookmarksFeedBuilder = (
       `bookmark."postId" = ${alias}.id AND bookmark."userId" = :userId`,
       { userId: ctx.userId },
     );
+
   if (unreadOnly) {
     newBuilder = newBuilder.andWhere((subBuilder) =>
       whereUnread(ctx.userId, subBuilder, alias),
     );
   }
-  if (listId && ctx.premium) {
-    newBuilder = newBuilder.andWhere('bookmark.listId = :listId', { listId });
+
+  const [firstFolderSubQuery, params] = getFirstFolderSubQuery({
+    builder: newBuilder,
+    userId: ctx.userId!,
+  });
+
+  if (ctx.isPlus) {
+    if (listId) {
+      newBuilder = newBuilder.andWhere('bookmark.listId = :listId', { listId });
+    } else {
+      newBuilder = newBuilder.andWhere('bookmark.listId IS NULL');
+    }
   }
+
+  if (!ctx.isPlus) {
+    if (listId) {
+      // Check if the list id is the first folder.
+      // This is when the user unsubscribed and trying to access locked folders.
+      newBuilder = newBuilder.andWhere(
+        `bookmark."listId" = (${firstFolderSubQuery})`,
```

Comment:
Can we query that `listId` once on top and then pass it here, feels like sub query will be inefficient on each row? Similar to what Chris said here https://github.com/dailydotdev/daily-api/pull/2507/files#r1875522680

Reply:
I'm open to either of the two, feel free to let me know on what you think is best.